app/code/Magento/Catalog/Model/Product/Gallery/CreateHandler.php

```diff
@@ -200,6 +202,10 @@ public function execute($product, $arguments = [])
         $this->processDeletedImages($product, $value['images']);
         $this->processNewAndExistingImages($product, $value['images']);
 
+        if (!$product->getStoreId()) {
+            $this->processSetImagesPositionToDefault($product->getEntityId(), $product->getStoreId());
```

Comment:
Beware. This should require a casting to `int`:

`$this->processSetImagesPositionToDefault((int) $product->getEntityId(), (int) $product->getStoreId());`

Unfortunately, `$product->getEntityId()` returns a string instead of `int`, but it is not correct to handle it as a string.